translator/translate/otel/receiver/jmx/translator.go

```diff
@@ -155,85 +130,105 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 		cfg.TruststorePath = truststorePath
 	}
 
-	if truststorePass, ok := jmxKeyMap[truststorePassword].(string); ok {
-		t.addRedactedMap(truststorePassword, truststorePass)
-		cfg.TruststorePassword = configopaque.String(truststorePass)
-	}
-
-	if truststoreType, ok := jmxKeyMap[truststoreType].(string); ok {
+	if truststoreType, ok := jmxKeyMap[truststoreTypeKey].(string); ok {
 		cfg.TruststoreType = truststoreType
 	}
 
-	if remoteProfile, ok := jmxKeyMap[remoteProfile].(string); ok {
+	if remoteProfile, ok := jmxKeyMap[remoteProfileKey].(string); ok {
 		cfg.RemoteProfile = remoteProfile
 	}
 
-	if realm, ok := jmxKeyMap[realm].(string); ok {
+	if realm, ok := jmxKeyMap[realmKey].(string); ok {
 		cfg.Realm = realm
 	}
 
-	if resourceAttributes, ok := jmxKeyMap[resourceAttributes].(map[string]interface{}); ok {
+	if resourceAttributes, ok := jmxKeyMap[resourceAttributesKey].(map[string]any); ok {
 		cfg.ResourceAttributes = convertToStringMap(resourceAttributes)
 	}
 
 	// set OTLP settings
 	cfg.OTLPExporterConfig.Endpoint = defaultOTLPEndpoint
-	if otlpMap, ok := jmxKeyMap[common.OtlpKey].(map[string]interface{}); ok {
+	if otlpMap, ok := jmxKeyMap[common.OtlpKey].(map[string]any); ok {
 		if endpoint, ok := otlpMap[common.Endpoint].(string); ok {
 			cfg.OTLPExporterConfig.Endpoint = endpoint
 		}
-		timeout, err := common.ParseDuration(otlpMap[timeout])
+		timeout, err := common.ParseDuration(otlpMap[otlpTimeoutKey])
 		if err == nil {
 			cfg.OTLPExporterConfig.Timeout = timeout
 		}
-		if headers, ok := otlpMap[headers].(map[string]interface{}); ok {
+		if headers, ok := otlpMap[otlpHeadersKey].(map[string]any); ok {
 			cfg.OTLPExporterConfig.Headers = convertToStringMap(headers)
 		}
 	}
 
-	return cfg, nil
+	var skipAuthValidation bool
+	if insecure, ok := jmxKeyMap[common.InsecureKey].(bool); ok {
+		skipAuthValidation = insecure
+	}
+
+	return cfg, validate(cfg, skipAuthValidation)
 }
 
-func convertToStringMap(input map[string]interface{}) map[string]string {
-	convertedMap := make(map[string]string)
-	for key, value := range input {
-		strKey := fmt.Sprintf("%v", key)
-		strValue := fmt.Sprintf("%v", value)
-		convertedMap[strKey] = strValue
+func validate(cfg *jmxreceiver.Config, skipAuthValidation bool) error {
+	if !skipAuthValidation && cfg.Endpoint != "" {
+		host, _, err := net.SplitHostPort(cfg.Endpoint)
+		if err != nil {
+			return fmt.Errorf("unable to parse endpoint: %w", err)
+		}
+		if !localhost.Contains(host) {
+			if err = validateAuth(cfg); err != nil {
+				return fmt.Errorf("jmx configuration with endpoint (%s): %w", cfg.Endpoint, err)
+			}
+		}
 	}
-	return convertedMap
+	return nil
 }
 
-func (t *translator) addRedactedMap(postfix string, value string) {
-	redactedMap[fmt.Sprintf("%v", t.factory.Type())+"/"+t.name+"/"+postfix] = value
+func validateAuth(cfg *jmxreceiver.Config) error {
+	var missingFields []string
+	for _, fields := range [][2]string{
```

Comment:
Can we make this array, into variable like `REQUIRED_FIELDS` so that we can use in testing , as well for easier editing.